src/client/app/+home/home.component.ts

```diff
@@ -13,45 +17,49 @@ import { GraphPreviewListService } from '../shared/index';
 
 export class HomeComponent implements OnInit {
 
-  newGraphPreview: string = '';
-  errorMessage: string;
-  graphPreviews: any[] = [];
+  userLoggedIn: boolean = false;
+  graphPreviews: DriveFile[] = [];
 
   /**
    * Creates an instance of the HomeComponent with the injected
    * GraphPreviewListService.
    *
-   * @param {GraphPreviewListService} GraphPreviewListService - The injected GraphPreviewListService.
+   * @param googleRealtimeService
+   * @param changeDetector
    */
-  constructor(public graphPreviewListService: GraphPreviewListService) {}
+  constructor(private googleRealtimeService: GoogleRealtimeService,
+              private changeDetector: ChangeDetectorRef,
+              private router: Router) {
+  }
 
   /**
    * Get the names OnInit
    */
   ngOnInit() {
-    this.getGraphPreviews();
+    this.googleRealtimeService.oauthToken.subscribe((oauthToken) => {
+      if (!!oauthToken) {
+        this.userLoggedIn = true;
+      }
+    });
+
+    this.googleRealtimeService.listFiles()
+        .subscribe((driveFiles: DriveFile[]) => {
+          this.graphPreviews = driveFiles;
+
+          // TODO(girum): Why do I need to call the change detector here?
+          this.changeDetector.detectChanges();
+        });
   }
 
-  /**
-   * Handle the nameListService observable
-   */
-  getGraphPreviews() {
-    this.graphPreviewListService.get()
-                     .subscribe(
-                       graphPreviews => this.graphPreviews = graphPreviews,
-                       error => this.errorMessage = <any>error
-                       );
+  authorize() {
+    this.googleRealtimeService.authorize(true);
   }
 
-  /**
-   * Pushes a new name onto the names array
-   * @return {boolean} false to prevent default form submit behavior to refresh the page.
-   */
-  addName(): boolean {
-    // TODO: implement nameListService.post
-    this.graphPreviews.push(this.newGraphPreview);
-    this.newGraphPreview = '';
-    return false;
+  createNewGraph() {
+    this.googleRealtimeService.createFile('Untitled uxgraph')
+        .subscribe((newGraph: DriveFile) => {
```

Comment:
Yeah, what you're describing is a Promise, the predecessor to Observables.

I think the idea with RxJS is that we should stick with Observables all the way around, to be able to compose and future-proof our asynchronous code, even in situations like this where this is a fire-once bit of async code.